pkg/cmd/cli/cmd/login/loginoptions.go

```diff
@@ -102,67 +107,70 @@ func (o *LoginOptions) getClientConfig() (*restclient.Config, error) {
 	o.Server = serverNormalized
 	clientConfig.Host = o.Server
 
```

Comment:
I think the old flow was easier to follow, though it could benefit from some refactoring. I think I'd prefer something like this. Note that we always attempt to pull in the CA first before dialing, dialToServer is stateless, it just uses the passed in clientConfig (and doesn't do CA searching on its own), and that the `o.InsecureTLS` check is only done when the error we encountered was an unknown authority issue (otherwise you could hit a nonexistent server and we'd happily continue on if you also passed `--insecure-skip-tls-verify`)

```
clientConfig := &restclient.Config{}

// load/normalize/prompt for server
clientConfig.Host = ...

// use specified CA or find existing CA
if len(o.CAFile) > 0 {
  clientConfig.CAFile = o.CAFile
  clientConfig.CAData = nil
} else {
  clientConfig.CAFile, clientConfig.CAData = findExistingClientCA(clientConfig.Host, *o.StartingKubeConfig)
}

// attempt to dial using the host/ca data in the client config
if err := dialToServer(clientConfig); err != nil {
  switch err.(type) {
  case x509.UnknownAuthorityError:
    if o.InsecureTLS {
      // set clientConfig to insecure mode
    } else {
      // prompt for insecure mode
    }
  case tls.RecordHeaderError:
    // return error about server not supporting HTTPS
  default:
    h1, p1, err1 := getHostPort(o.Server)
    h2, p2, err2 := getHostPort(defaultClusterURL)
    if err1 == nil && err2 == nil && p1 != p2 {
      // return error, hinting at using different port
    }
    // return error
  }
}
```